c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -1128,6 +1129,34 @@ JSString *method(JSContext *cx, HandleObject obj) {
   return JS::GetReservedSlot(obj, Slots::Method).toString();
 }
 
+bool set_cache_key(JSContext *cx, HandleObject self, HandleValue cache_key_val) {
+  MOZ_ASSERT(is_instance(self));
+  size_t key_len;
+  // Convert the key argument into a String following https://tc39.es/ecma262/#sec-tostring
+  JS::UniqueChars keyString = encode(cx, cache_key_val, &key_len);
+  if (!keyString) {
+    return false;
+  }
+  std::string key(keyString.get(), key_len);
```

Comment:
Does `key` need to copy the bytes from `keyString`, or would it be possible to make a `std::string_view` and use that with `hash256_hex_string` instead?

Reply:
I've changed to string_view 👍